### PR TITLE
Fix ruff errors in routes public job

### PR DESCRIPTION
**Pull request by gopidesupavan**
https://github.com/apache/airflow/actions/runs/12013016097/job/33486044971?pr=44349#step:8:336

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/api_fastapi/core_api/routes/public/job.py [hunk 1]
```diff
@@ -16,10 +16,11 @@
 # under the License.
 from __future__ import annotations
 
+from typing import Annotated
+
 from fastapi import Depends, status
 from sqlalchemy import select
 from sqlalchemy.orm import Session
-from typing_extensions import Annotated
```
Comment (gopidesupavan) [inline airflow/api_fastapi/core_api/routes/public/job.py:22]: Not sure is it use `typing_extensions` or `typing`. pre-commit failing. if it is `typing_extensions` is required may be we have to add ignore?

Comment (pierrejeambrun) [reply airflow/api_fastapi/core_api/routes/public/job.py:22]: This is right, it should come from Annotated.

**Review by pierrejeambrun — APPROVED** [review]
Thanks

**Comment by pierrejeambrun** [comment]
I'm surprised to see that the original build for the PR associated to it was successful. (Ruff didn't complain):
https://github.com/apache/airflow/actions/runs/11983768727/job/33477766501?pr=43859

**Comment by gopidesupavan** [comment]
> I'm surprised to see that the original build for the PR associated to it was successful. (Ruff didn't complain): https://github.com/apache/airflow/actions/runs/11983768727/job/33477766501?pr=43859

ah might be due to recent upgrade to ruff version :)

**Comment by gopidesupavan** [comment]
> > I'm surprised to see that the original build for the PR associated to it was successful. (Ruff didn't complain): https://github.com/apache/airflow/actions/runs/11983768727/job/33477766501?pr=43859
> 
> ah might be due to recent upgrade to ruff version :)

https://github.com/apache/airflow/pull/44287

**Comment by pierrejeambrun** [comment]
Got it thanks.

**Pull request merged**